### PR TITLE
Add enemies and bug fix

### PR DESCRIPTION
**Pull request by vchadha**
bug fix for initial scene name
basic prefabs for very basic enemies

When adding an enemy to a scene:
create an empty (or multiple) with the path tag
for each path you can add more empties and tag them as waypoints

Then we will go through each path and move along it one at a time.
Supports reverse, restart etc.

Kinda cool cuz you can see the path change for each enemy as it iterates through the paths: see video (paths aren't final)

https://github.com/justinro-underscore/DinoJam2023/assets/28954155/bec87101-43ad-411a-ba3f-194f0f27abc9

### CODE REVIEW

**Review by justinro-underscore — CHANGES_REQUESTED** [review]
Eh I'm not feelin it

**Comment by vchadha** [comment]
> Eh I'm not feelin it

son of a...... 😆 

but also i didn't like this before so changed it lmao

File: Assets/Scripts/GameElements/EnemyMovementController.cs [hunk 1]
```diff
@@ -3,37 +3,170 @@
...
+
+    protected override void ManagedStart()
+    {
+        // Init list of paths
+        paths = new List<Path>();
+
+        // Find child objects with path tag
+        foreach (Transform child in transform)
+        {
+            if (child.tag == Constants.pathTag)
```
Comment (justinro-underscore) [inline Assets/Scripts/GameElements/EnemyMovementController.cs]: This is minor, but I think in Unity usually when comparing tags you want to use the `child.CompareTag(Constants.pathTag)` function

File: Assets/Scripts/GameElements/EnemyMovementController.cs [hunk 2]
```diff
@@ -3,37 +3,170 @@
...
+            {
+                // Find sub children with waypoint tag
+                // ie. paths have waypoints
+                List<Vector3> vectorWaypoints = new List<Vector3>();
+                foreach (Transform subChild in child)
+                {
+                    if (subChild.tag == Constants.waypointTag)
```
Comment (justinro-underscore) [inline Assets/Scripts/GameElements/EnemyMovementController.cs]: Same here!

File: Assets/Scripts/GameElements/EnemyMovementController.cs [hunk 3]
```diff
@@ -3,37 +3,170 @@
 
 using DG.Tweening;
 
-public class EnemyMovementController : MonoBehaviour {
+public class EnemyMovementController : IManagedController
+{
+    // Struct for paths
+    private struct Path
```
Comment (justinro-underscore) [inline Assets/Scripts/GameElements/EnemyMovementController.cs]: I think Unity doesn't really like structs all that much. I'd recommend making this a class

**Review by justinro-underscore — CHANGES_REQUESTED** [review]

**Review by justinro-underscore — APPROVED** [review]
Meant to approve! 💩

File: Assets/Scripts/GameElements/EnemyMovementController.cs (hunk 3 above)
Comment (vchadha) [reply Assets/Scripts/GameElements/EnemyMovementController.cs]: Unity whyyyyyy

**Pull request merged**